File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.cpp
class Solution {
public:
    vector<vector<int>> findWinners(vector<vector<int>>& matches) {
        map<int, pair<int,int>> mp;
        for(auto it : matches) {
            mp[it[0]] = {mp[it[0]].first+1, mp[it[0]].second};
            mp[it[1]] = {mp[it[1]].first, mp[it[1]].second+1};
        }

        vector<vector<int>> ans(2);
        for(auto it : mp) {
            if(it.second.second == 0) ans[0].push_back(it.first);
            if(it.second.second == 1) ans[1].push_back(it.first);
        }

        return ans;
    }
};

File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.java
class Solution {
    public List<List<Integer>> findWinners(int[][] matches) {
        Map<Integer, Integer> map = new TreeMap<>();
        for(int[] it : matches) {
            map.put(it[0], map.getOrDefault(it[0], 0));
            map.put(it[1], map.getOrDefault(it[1], 0) + 1);
        }

        List<List<Integer>> ans = new ArrayList<>(2);
        for(int i = 0; i<2; i++) {
            ans.add(new ArrayList<>());
        }

        for(Integer player : map.keySet()) {
            if(map.get(player) <= 1) {
                ans.get(map.get(player)).add(player);
            }
        }

        return ans;
    }
}